### ISSUE TITLE
author report filter

### ISSUE THREAD

**Issue by alishaevn** [scientist-softserv/palni-palci#721]
# Story
update the following reports to respond to the `author` query param.
https://countermetrics.stoplight.io/docs/counter-sushi-api/yc1o09env0d2a-counter-sushi-api

- [ ] Item report

# Acceptance Criteria

- [ ] Item report can be filtered by the `author`.

# Screenshots / Video

# Testing Instructions and Sample Files
**Developer to include URLs with the proper filters for testing**
- [ ] View the Item report in browser

**Comment by alishaevn** [comment]
blocked by #695

**Comment by jeremyf** [comment]
Do we know what Fedora properties we should count as `author`?

**Comment by alishaevn** [comment]
> Do we know what Fedora properties we should count as author?

@jeremyf my understanding is that we are pointing it at `creator`.
ref: [slack thread](https://assaydepot.slack.com/archives/C05HUG89YVD/p1692631643844159?thread_ts=1692137277.543219&cid=C05HUG89YVD).

**Comment by alishaevn** [comment]
for additional reference:

<details>
<summary>another slack thread</summary>

![Image](https://github.com/scientist-softserv/palni-palci/assets/29032869/137b3e61-8a92-4752-b262-7628ff58037a)
</details>

**Comment by alishaevn** [comment]
this ticket is still blocked by #695. the column was added to the `hyrax_counter_metrics` table, but the`ImportCounterMetrics` class hasn't been updated to use it.

**Comment by summer-cook** [comment]
@labradford I think that is because none of the data in the pittir.commons-archive.org have an author with that name (i just searched, and it seems like every item has an author of "null" in that tenant?)

I will check it on another tenant that has some authors

**Comment by summer-cook** [comment]
hmmm ok, just tested it again at 

https://dev.palni-palci-staging.notch8.cloud/api/sushi/r51/reports/ir?begin_date=05-2023&end_date=08-202&author=Summer

there is one work in the db with a creator of "Summer" haha. it did not find that one work

**Comment by alishaevn** [comment]
> QA - getting this error on pittir.commons-archive.org "error": "The given parameter `author=Hamilton, Carol V.` was malformed. Please provide the first author's name exactly as it appears on the work."

agreed with summer that the carol hamilton example was just an example. that's a name I had in local dev. above that example in the instructions section it said to view all items first, find an author, then filter by that author name. when viewing https://dev.commons-archive.org/api/sushi/r51/reports/ir?begin_date=2023-06&end_date=2023-06, each author name shows as null. so this is a data on staging issue, not a filter issue.

since summer saw a work with a creator in the db, perhaps the rake task hasn't been re-run on staging to update the `author` field on the `Hyrax::CounterMetric` items?

<details>
<summary>https://dev.commons-archive.org/api/sushi/r51/reports/ir?begin_date=2023-06&end_date=2023-06</summary>

<img width="1008" alt="image" src="https://github.com/scientist-softserv/palni-palci/assets/29032869/3ee1cfd0-4782-4b41-8178-e82eefa3f022">

</details>

**Comment by summer-cook** [comment]
I just re-ran it on the dev tenant on palni-palci-staging, after creating a work with an author with a strangely formatted name as well, and it was able to find it when i typed it in exactly like so 


![Image](https://github.com/scientist-softserv/palni-palci/assets/73361970/432f1aa1-dfeb-4607-836f-7c0d059e8ffe)

**Comment by summer-cook** [comment]
I think its because none of the works in the pittir.commonsarchive.org tenant have authors. ill add some and rerun the rake task

**Comment by alishaevn** [comment]
if the same csv we used locally with 5 works was imported on staging, then yes, none of those works will have the `creator` field filled out because the creator related columns on that csv do not align with what bulkrax is looking for.

either the works have to be manually updated to have a creator, or the csv below can be imported to update them. (just those 5)
[pittir-import-works-with-creator-column.csv](https://github.com/scientist-softserv/palni-palci/files/12440835/pittir-import-works-with-creator-column.csv)

**Comment by ctgraham** [comment]
This will primarily be of interest in the future if we add identifiers such as ORCID to the Creators.  Does the underlying datastructure potentially scale to this, were additional development to be undertaken?

**Comment by alishaevn** [comment]
> This will primarily be of interest in the future if we add identifiers such as ORCID to the Creators. Does the underlying datastructure potentially scale to this, were additional development to be undertaken?

@ctgraham as of now, the `Hyrax::CounterMetric` database table includes a column of "author" that expects a string, according to the counter specs. do you mean that you may wish to instead store the ORCID of a creator in that column instead of a name?

**Comment by ctgraham** [comment]
I anticipate a future enhancement where the Creator is actually an RDF link to an internal or external taxonomy.  It is unlikely that a repository will be able to provide an ORCID for every creator, but I could imagine a Creator which has links to and ORCID ID, an institutional id, a Scopus ID, etc.

If that were to happen, how baked-in is the assumption that the `Hyrax::CounterMetric` can only ever have one and only one author identifier?

**Comment by alishaevn** [comment]
> If that were to happen, how baked-in is the assumption that the Hyrax::CounterMetric can only ever have one and only one author identifier?

it’s pretty baked in, but not too large of an undertaking to change course if we create a new sow for it in the future.

**Comment by alishaevn** [comment]
@ctgraham can you re-review this? we began accounting for multiple authors 2 weeks ago with [this pr](https://github.com/scientist-softserv/palni-palci/pull/771).

**Comment by ctgraham** [comment]
I tried queries of:
https://dev.palni-palci-staging.notch8.cloud/api/sushi/r51/reports/ir?begin_date=2023-06&end_date=2023-07
and
https://pittir.commons-archive.org/api/sushi/r51/reports/ir?begin_date=2023-06&end_date=2023-07

But didn't see any results with multiple authors.  What data can I use for QA?

**Comment by jeremyf** [comment]
I made some changes to data regarding the entries authored by "Summer".  There are now two authors: Summer and Winter.

**Comment by ctgraham** [comment]
# Pitt QA concerns

When requesting multiple authors with pipes, the response in inconsistent.  For example
https://dev.palni-palci-staging.notch8.cloud/api/sushi/r51/reports/ir?begin_date=2023-06&end_date=2023-07&author=Summer|Winter
Returns data.

https://dev.palni-palci-staging.notch8.cloud/api/sushi/r51/reports/ir?begin_date=2023-06&end_date=2023-07&author=Winter|Summer
Returns an error.

The expected behavior is for both of these to return an error, as the author filter doesn't describe pipe separation capability.

**Comment by jeremyf** [comment]
According to the Sushi spec, the Item Report does not account for multiple authors.

https://countermetrics.stoplight.io/docs/counter-sushi-api/5a6e9f5ddae3e-ir-item-report

> Identifier of a specific author usage is being requested for. If omitted, all items on the platform with usage for the customer will be returned.

Later, for `yop` there is language about how to encode multiple values.

Which, I believe means that if we include a `|` in the author we should return a 3060.  So I'll set about working on that.  (That does assume a person does not have a `|` in their name).  Note, the data encoding we're doing is to have pipe separation of multiple authors in the database.  The record you're seeing has `|Summer|Winter|` for a multi-author file.  And there's a direct match.

**Comment by alishaevn** [comment]
@ctgraham this is available for review again. both links that you provided above are now returning a 3060 exception per jeremy's explanation above.

**Comment by ctgraham** [comment]
The data line appears to be missing a closing paren:
"You may not query for multiple authors (as specified by the `|' delimiter."

Otherwise, works as expected.

**Comment by alishaevn** [comment]
@ctgraham this is ready for review: https://dev.commons-archive.org/api/sushi/r51/reports/ir?begin_date=2023-06&end_date=2023-07&author=Summer|Winter

**Comment by ctgraham** [comment]
Confirmed.
